mlxtend/feature_selection/sequential_feature_selector.py

```diff
@@ -393,188 +460,162 @@ def fit(self, X, y, custom_feature_names=None, groups=None, **fit_params):
                     " than the max k_features value."
                 )
 
-        if isinstance(self.k_features, tuple) or isinstance(self.k_features, str):
-
-            select_in_range = True
-
-            if isinstance(self.k_features, str):
-                if self.k_features not in {"best", "parsimonious"}:
-                    raise AttributeError(
-                        "If a string argument is provided, "
-                        'it must be "best" or "parsimonious"'
-                    )
-                else:
-                    min_k = 1
-                    max_k = X_.shape[1]
-            else:
-                min_k = self.k_features[0]
-                max_k = self.k_features[1]
-
-        else:
-            select_in_range = False
-            k_to_select = self.k_features
+        is_parsimonious = False
+        if isinstance(self.k_features, str):
+            if self.k_features not in {"best", "parsimonious"}:
+                raise AttributeError(
+                    "If a string argument is provided, "
+                    'it must be "best" or "parsimonious"'
+                )
+            if self.k_features == "parsimonious":
+                is_parsimonious = True
 
-        orig_set = set(range(X_.shape[1]))
-        n_features = X_.shape[1]
+        min_n_groups = len(self.fixed_features_group_set)
+        max_n_groups = len(self.feature_groups)
+        if isinstance(self.k_features, str):
+            self.k_features = (min_n_groups, max_n_groups)
+        elif isinstance(self.k_features, int):
+            # we treat k_features as k group of features
+            self.k_features = (self.k_features, self.k_features)
```

Comment:
Maybe avoid over-writing `self.k_features`. And, instead, use a new variable, say, `k_range` (better to avoid using `feature` in the name of variable because it is in fact the range of groups when `feature_groups` is provided.

Reply:
Now, we use `k_min_and_max`.